### DISCUSSION TITLE
Specify the number of speakers

### DISCUSSION THREAD

**Discussion by a-kifumi** [deepgram/community#176] [Feature Requests] [answered]
Used in Japanese, the results are returned as if there were three speakers even though there are only two speakers.

I would like to limit the number of speakers by specifying options such as max_speaker.

<img width="659" alt="スクリーンショット 2023-05-30 15 12 36" src="https://github.com/deepgram/community/assets/8614839/25ec5235-8fa1-4851-8498-7624f036ecd8">

**Comment by jjmaldonis** [comment, marked as answer]
Hey @yoshisansan! We do not currently have support for setting the max number of speakers, but it's likely that this will be supported in the future or the diarization results will improve so dramatically that it won't be needed as we continue to improve our models. Please keep an eye out for an email or updated docs with support for `max_speakers`. If/when that feature becomes available, we'll try to update this thread.

**Reply by a-kifumi** [reply]
I see. I look forward to it.

**Comment by erikmunson** [comment]
big +1 to this, I posted a note about why this is valuable on a different thread that seems to be a dupe of this one: https://github.com/orgs/deepgram/discussions/217#discussioncomment-7087013

**Comment by mehdigreefhorst** [comment]
+1 this is very needed, it often gets it wrong currently if you only have a single channel

**Comment by onurozgurozkan** [comment]
Is there any news about that feature?

**Reply by jjmaldonis** [reply]
Hey @onurozgurozkan, no ETA at the moment but I just sent your feedback to our team.

**Comment by arsi-rad** [comment]
+1 we find that diarization is unusable because it returns incorrect number of speakers, but we know how many speakers in our use case

**Comment by leafcutter-au** [comment]
+1 for me as well.  My application would require a max speakers parameter as the ever growing number of speakers in the current transcription is problematic.  Do we have an idea if this is still a planned feature?

**Comment by sthorat789** [comment]
+1 for me as well.
